kili/mutations/asset/helpers.py

```diff
@@ -74,12 +77,29 @@ def process_content(input_type: str,
     if input_type in ['IMAGE', 'PDF']:
         return [content if is_url(content) else (content
             if (json_content_array is not None and json_content_array[i] is not None)
-                else encode_base64(content))
+            else (encode_base64(content) if check_file_mime_type(content, input_type) else None))
             for i, content in enumerate(content_array)]
     if input_type == 'FRAME' and json_content_array is None:
-        content_array = list(map(encode_object_if_not_url, content_array))
+        content_array = [encode_object_if_not_url(content, input_type) for content in content_array]
     return content_array
 
+def check_file_mime_type(content: str, input_type: str):
+    """
+    Returns true if the mime type of the file corresponds to the allowed mime types of the project
+    """
+    if input_type in ['IMAGE', 'FRAME', 'PDF']:
+        mime_type = get_data_type(content.lower())
+        if mime_extensions_for_IV2[input_type] and mime_type:
+            correct_mime_type = mime_type in mime_extensions_for_IV2[input_type]
+            if not correct_mime_type:
+                print(f'File mime type for {content} is {mime_type} and does not correspond' \
+                    'to the type of the project. '\
+                    f'File mime type should be one of {mime_extensions_for_IV2[input_type]}')
+            return correct_mime_type
+        if not (mime_extensions_for_IV2[input_type] and mime_type):
```

Comment:
The if is not necessary, you can return False directly

Reply:
Also added as an early return